dashboard/common/db_util.py

```diff
@@ -195,3 +196,51 @@ def get_canvas_data_date() -> Union[datetime, None]:
     except Exception:
         logger.info("Value could not be found from metadata", exc_info = True)
     return None
+
+def map_dict_to_query_job_config(param_dict: Dict[str, Any]) -> bigquery.QueryJobConfig:
```

Comment:
Just curious, why is this function needed? with postgres this was not needed.

Reply:
This is needed because BigQuery has special types in it's JobConfig that have to be passed explicitly to bind parameters. 

All of the bindings could have been changed and this would have been unneeded but I wasn't sure which method would have been easier. This is more compatibility for the old code. For instance I've got a call like this which isn't changed much from the old code.

```
        query_params = {
                        'course_ids': self.valid_locked_course_ids,
                        'canvas_data_id_increment': settings.CANVAS_DATA_ID_INCREMENT,
                        }
        df = self.execute_bq_query(self.queries['submission'], query_params).to_dataframe()
```

This could have been rewritten completely in BigQuery style to pass a `QueryJobConfig `object explicitly. 

```
job_config = bigquery.QueryJobConfig(
    query_parameters=[
        bigquery.ArrayQueryParameter('course_ids', 'INT64', self.valid_locked_course_ids,),
        bigquery.ScalarQueryParameter('canvas_data_id_increment', 'INT64', settings.CANVAS_DATA_ID_INCREMENT),
    ]
)

df = self.execute_bq_query(self.queries['submission'], query_params).to_dataframe()
```

Perhaps this would have just been a better idea to do this? I could still do this and remove this helper function as there's only 5 places this was done. I thought it would have been more work to change and it would have to support PostGres syntax still, but that was before this all changed fully to BigQuery. It just requires you to learn more about the special BigQuery types rather than infer them.

Reply:
I have made this change in the latest version of the code.